src/mongocxx/test/client.cpp

```diff
@@ -73,6 +73,68 @@ TEST_CASE("A client lists its databases with a filter applied", "[client]") {
     REQUIRE(client_list_databases_called);
 }
 
+TEST_CASE("list databases with a filter works", "[client]") {
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+
+    instance::current();
+
+    mongocxx::client client{mongocxx::uri{"mongodb://localhost:27017"}};
+    // auto filter_doc = make_document(kvp("filter", make_document(kvp("name", "admin"))));
+    auto cursor = client.list_databases();
+    for (auto&& doc : cursor) {
+        std::cout << bsoncxx::to_json(doc) << std::endl;
+    }
+}
+
+TEST_CASE("list databases passes authorizedDatabases option", "[client]") {
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+    using bsoncxx::to_json;
+
+    auto mock_client_find_databases_with_opts =
+        mongocxx::libmongoc::client_find_databases_with_opts.create_instance();
+    int num_called = 0;
```

Comment:
It seems like this is used to assert that `mock_client_find_databases_with_opts` was called. I think there's already `visit` function you could override for functions declared in `helpers.hpp`. An example is on line 177-178: 
```
    bool destroy_called = false;
    client_destroy->visit([&](mongoc_client_t*) { destroy_called = true; });
```